syntax = "proto3";

package consumption;

service ConsumptionService {
  // Get electricity consumption data, optionally filtered by date range
  rpc GetConsumptionData (ConsumptionRequest) returns (ConsumptionResponse);
}

message ConsumptionRequest {
  string start_datetime = 1; // Optional, ISO 8601 format
  string end_datetime = 2;   // Optional, ISO 8601 format
}

message ConsumptionRecord {
  string datetime = 1;
  double energy_usage = 2;
  double temperature = 3;
}

message ConsumptionResponse {
  repeated ConsumptionRecord records = 1;
}